packages/web-app-files/tests/unit/components/AppBar/AppBar.spec.js

```diff
@@ -0,0 +1,360 @@
+        wrapper = getShallowWrapper(route, store)
+      })
+
+      it('should show "New" button', () => {
+        const newButton = wrapper.find(elSelector.newFileButton)
+        const sizeInfo = wrapper.find(elSelector.sizeInfo)
+
+        expect(sizeInfo.exists()).toBeFalsy()
+        expect(newButton.isVisible()).toBeTruthy()
+        expect(newButton.props('ariaLabel')).toEqual('Add files or folders')
+      })
+      it('should show default file menu items', () => {
+        const ocDrop = wrapper.find(elSelector.ocDrop)
+        const fileUpload = wrapper.find(elSelector.fileUpload)
+        const folderUpload = wrapper.find(elSelector.folderUpload)
+        const newFolderBtn = wrapper.find(elSelector.newFolderBtn)
+        const fileMenuList = wrapper.findAll(elSelector.fileMenuList)
+
+        expect(ocDrop.isVisible()).toBeTruthy()
+        expect(fileUpload.isVisible()).toBeTruthy()
+        expect(folderUpload.isVisible()).toBeTruthy()
+        expect(newFolderBtn.isVisible()).toBeTruthy()
+        expect(fileMenuList.length).toBe(3)
+      })
+      it('should trigger "showCreateResourceModal" if new-folder button is clicked', async () => {
+        const store = createStore({ currentFolder, selected: [] })
+        wrapper = getWrapper(route, store)
+
+        const newFolderBtn = wrapper.find(elSelector.newFolderBtn)
+        await newFolderBtn.trigger('click')
+
+        expect(spyShowCreateResourceModal).toHaveBeenCalled()
+      })
+      it('should show extra file menu items', () => {
+        const store = createStore({ currentFolder, selected: [] }, newFileHandlers)
+        wrapper = getShallowWrapper(route, store)
+        const newTextFileBtn = wrapper.find(elSelector.newTextFileBtn)
+        const newMdFileBtn = wrapper.find(elSelector.newMdFileBtn)
+        const newDrawioFileBtn = wrapper.find(elSelector.newDrawioFileBtn)
+        const fileMenuList = wrapper.findAll(elSelector.fileMenuList)
+
+        expect(newTextFileBtn.isVisible()).toBeTruthy()
+        expect(newMdFileBtn.isVisible()).toBeTruthy()
+        expect(newDrawioFileBtn.isVisible()).toBeTruthy()
+        expect(fileMenuList.length).toBe(6)
+      })
+      it.each(newFileHandlers)(
+        'should trigger "showCreateResourceModal" if new file button is clicked',
+        async fileHandler => {
+          const store = createStore({ currentFolder, selected: [] }, newFileHandlers)
+          wrapper = getWrapper(route, store)
+
+          const button = wrapper.find(getFileHandlerSelector(fileHandler.ext))
+          await button.trigger('click')
+
+          expect(spyShowCreateResourceModal).toHaveBeenCalled()
+          expect(spyShowCreateResourceModal).toHaveBeenCalledWith(
+            false,
+            fileHandler.ext,
+            fileHandler.action
+          )
+        }
+      )
+    })
+
+    describe('when an item is selected', () => {
+      it('should show size info', () => {
```

Comment:
you've tested that new but is not visible when items are not selected. can the counterpart go inside this describe?